RIP-3 Nft Bank/RIP-3 Nft Bank.md

```diff
@@ -0,0 +1,81 @@
+
+Users may also launch their own contracts, which they will have full control over. Though expensive to do, they need not fear that the contracts, and therefore their assets, will be frozen when bugs/security risks are discovered.
+
+### Proposed Technical Design
+
+#### LibBank.sol-
+Structure object:
+-   AssetData
+    -   address contract address
+    -   uint256 tokenId
+-   SecurityInfo
+    -   uint64 lastAccessed - blockNumber of the last time an account was accessed.
+    -   bool accountLocked - whether an account is locked or not
+    -   uint64 backupCounter
+    -   address[] backupAddresses- backUp accounts in case the main account's private keys are lost
+#### Bank.sol-
+Inherits:
+-   ERC1155HolderUpgradeable and ERC721HolderUpgradeable which gives the contract the ability to receive and store NFTs.
+-   IERC721Upgradeable and IERC1155Upgradeable which allows the contract to make transfer calls.
+-   ERC165CheckerUpgradeable which gives the contract the ability to check supported interfaces.
+-   PausableUpgradeable which provides the functionality of locking functions in case a bug is discovered.
+-   IBank and IBankStorage interfaces.
+
+Global variables:
+-   accountInfo (address => LibBank.SecurityInfo)
+-   uint64 averageBlockTime
+-   uint64 lastPaused - blockNumber of when the contracts have been frozen
+
+Functions:
+
+Note: All mutative functions will have a modifier that requires that either msg.sender == the affected account's address or that the current blockNumber - lastAccessed variable is over a certain time frame and that msg.sender == one of the backup addresses. And all mutative functions will reset the lastAccessed blockNumber of an address to the current blockNumber, but not if the functions were accessed by the backup addresses.
+
+-   depositBatch function to deposit multiple assets. This function takes in an array of assetData structs, an array of uint256 deposit amounts and an address.
+    -   checks if the user possesses the amount of tokens they want to deposit.
+    -   calls the deposit function in BankStorage.sol to update the storedAmounts mapping.
+    -   calls a transfer function to send the item to Bank.sol.
+-   withdrawBatch function to withdraw multiple assets. This function takes in an array of assetData structs, an array of uint256 withdraw amounts and an address.
+    -   requires that stored amount of the item is greater than or equal to the amount to withdraw.
+    -   calls the withdraw function in BankStorage.sol to update the storedAmounts mapping.
+    -   calls a transfer function to retrieve the item from the bank.
+-   addBackup function to add a back up address in case one loses their keys.
+-   removeBackup function to remove an address from the backupAddresses array.
+-   lock function to set the accountLocked boolean to true or false.
```

Comment:
this one should come with a password and a signature. I'll show you later how that's verified